### DISCUSSION TITLE
변수 생성의 중요성....(feat 삽질!!!)

### DISCUSSION THREAD

**Discussion by YurimYang** [SCBJ-7/SCBJ-BE#57] [Q&A]
### 발생과정

![image](https://github.com/SCBJ-7/SCBJ-BE/assets/63856521/c9c04e9e-fb50-4a0f-af0e-1bd993baaeaa)
- 양도글 service 로직을 작성 중에 계속해서 productId가 null이란 exception을 발견했다.
- 다만 service 로직은 틀리지 않을 것이라 생각에 nullable=false인 entity를 build하지 않아서 생기는 문제라 파악했다.

### 원인

- 하지만 진짜 원인은,,!!!! 단순히 save한걸 변수에 담지 않고 바로 return해서 생기는 문제였다.

### 해결방안
![image](https://github.com/SCBJ-7/SCBJ-BE/assets/63856521/3775f26b-eca2-4380-9b70-5392715d6d64)

- @Kwonminwoo 님 덕분에 변수에 담지 않음을 확인했고, 이를 수정하여 고칠 수 있었습니다 👍

**Comment by Kwonminwoo** [comment]
와우 기록 굿굿 👍 까먹지 않으시겠네요!
